compiler/src/iree/compiler/Codegen/Common/GPU/GPUDistributionPatterns.cpp

```diff
@@ -597,6 +597,50 @@ struct DistributeScfFor final : OpDistributionPattern<scf::ForOp> {
   }
 };
 
+struct DistributeTranspose final : OpDistributionPattern<vector::TransposeOp> {
+  using OpDistributionPattern::OpDistributionPattern;
+
+  LogicalResult matchAndRewrite(vector::TransposeOp transposeOp,
+                                DistributionSignature &signature,
+                                PatternRewriter &rewriter) const override {
+    VectorValue value = transposeOp.getVector();
+    LayoutAttr layout = dyn_cast<LayoutAttr>(signature[value]);
+    if (!layout) {
+      return failure();
+    }
+
+    /// Transpose only changes the notion of where the data carried by each
+    /// thread comes from in the SIMD vector. The data carried by each thread is
+    /// still the same, just iterated in a new permuted order. This iteration
+    /// information is carried by the layout. So, we can simply distribute
+    /// transpose to a no-op. Example:
```

Comment:
The layout carries the order of dimensions from fastest changing to slowest changing.

Reply:
Ok I see, this might need some revisiting for vector level transposes then (i.e. say transfer_write just takes whatever layout the transpose gives, could end up producing some non-optimal writes by having vector_x be outermost), but those aren't very common so we can worry about that later.

Reply:
Yeah this is something we need to take care about. This would be generally solved by setting a layout on write.